lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -708,19 +709,18 @@ func installKubeAgent(ctx context.Context, cfg installKubeAgentParams) error {
 	if cfg.req.IsCloud && cfg.req.EnableAutoUpgrades {
 		vals["updater"] = map[string]any{"enabled": true, "releaseChannel": "stable/cloud"}
 
-		vals["highAvailability"] = map[string]any{"replicaCount": 2,
+		vals["highAvailability"] = map[string]any{
+			"replicaCount":        2,
 			"podDisruptionBudget": map[string]any{"enabled": true, "minAvailable": 1},
 		}
 	}
 	if modules.GetModules().BuildType() == modules.BuildEnterprise {
 		vals["enterprise"] = true
 	}
 
-	eksTags := make(map[string]*string, len(cfg.eksCluster.Tags))
-	for k, v := range cfg.eksCluster.Tags {
-		eksTags[k] = aws.String(v)
-	}
-	eksTags[types.OriginLabel] = aws.String(types.OriginCloud)
+	eksTags := maps.Clone(cfg.eksCluster.Tags)
+
+	eksTags[types.OriginLabel] = types.OriginCloud
```

Comment:
Look that maps.Clone doesn't create a new map if the src map is nill. Use make(map[string]string) followed by maps.Copy otherwise the last line will panic

Reply:
Nice catch!